### ISSUE TITLE
How to construct a floatgrid by cube. sphere or cone?

### ISSUE THREAD

**Issue by hjhj111** [AcademySoftwareFoundation/openvdb#1283]
I can construct a floatgrid from triangle meshes, but how to construct a floatgrid by cube. sphere or cone?

**Comment by cgolchert** [comment]
I'm not sure if you are talking about csg versions of not but you could
look at the platonic solids files to see how it feeds those into grids.  It
will at least, directly, give you a cube.

**Comment by hjhj111** [comment]
Thank you. I know how to get a float grid of levelset from triangle meshes. But if the input was a simple 3D primitives like a cube , cone, cylinder or so. I am not clear about what you said about the files

**Comment by hjhj111** [comment]
Yes, I get the file but it seems that just platonic solids are surppoted.

**Comment by hjhj111** [comment]
Is this mean I have to use triangles to represent a model  even if it was a simple primitive that can be represent by a sdf function

**Comment by apradhana** [comment]
For cube, you can use [createLevelSetPlatonic](https://github.com/AcademySoftwareFoundation/openvdb/blob/master/openvdb/openvdb/tools/LevelSetPlatonic.h#L65). 

For sphere, you can use [createLevelSetSphere](https://github.com/AcademySoftwareFoundation/openvdb/blob/b58e5767a9e5f6dfe0dd2194828b624c01ce4dd9/openvdb/openvdb/tools/LevelSetSphere.h#L50). 

For a cone, you can construct a tri-mesh from your primitive and then invoke [meshToVolume](https://github.com/AcademySoftwareFoundation/openvdb/blob/a47b9c36b6c840c87a8ae93c61aa0cb95698b19c/openvdb/openvdb/tools/MeshToVolume.h#L113). Another option is to write a level-set for a cone yourself using a [cone signed distance formula](https://www.iquilezles.org/www/articles/distfunctions/distfunctions.htm) in a functor.
